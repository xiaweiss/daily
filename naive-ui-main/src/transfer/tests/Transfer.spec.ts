import { mount } from '@vue/test-utils'
import { sleep } from 'seemly'
import { NTransfer } from '../index'

describe('n-transfer', () => {
  it('should work with import on demand', () => {
    mount(NTransfer)
  })

  it('should work with `disabled` prop', () => {
    const wrapper = mount(NTransfer, { props: { disabled: true } })
    expect(wrapper.find('.n-transfer').attributes('class')).toContain(
      'n-transfer--disabled'
    )
  })

  it('should work with `filterable` prop', () => {
    const wrapper = mount(NTransfer, { props: { filterable: true } })
    expect(wrapper.find('.n-transfer').attributes('class')).toContain(
      'n-transfer--filterable'
    )
  })

  it('should work with `filter` prop', async () => {
    const options = [
      {
        label: 'test1',
        value: 'test1'
      }
    ]
    const onFilter = jest.fn()
    const wrapper = mount(NTransfer, {
      props: { filterable: true, filter: onFilter, options: options }
    })
    await wrapper.find('input').setValue('1')
    await sleep(300)
    expect(onFilter).toHaveBeenCalled()
  })

  it('should work with `size` prop', async () => {
    ;(['small', 'medium', 'large'] as const).forEach((i) => {
      const wrapper = mount(NTransfer, {
        props: { size: i }
      })
      expect(wrapper.find('.n-transfer').attributes('style')).toMatchSnapshot()
    })
  })

  it('should work with `source-filter-placeholder`、`target-filter-placeholder` props', async () => {
    const wrapper = mount(NTransfer, {
      props: {
        filterable: true,
        'source-filter-placeholder': 'test-source',
        'target-filter-placeholder': 'test-target'
      }
    })
    expect(wrapper.findAll('input')[0].attributes('placeholder')).toBe(
      'test-source'
    )
    expect(wrapper.findAll('input')[1].attributes('placeholder')).toBe(
      'test-target'
    )
  })

  it('should work with `source-title`、`target-title` props', async () => {
    const wrapper = mount(NTransfer, {
      props: {
        'source-title': 'test-source',
        'target-title': 'test-target'
      }
    })
    expect(wrapper.findAll('.n-transfer-list-header__header')[0].text()).toBe(
      'test-source'
    )
    expect(wrapper.findAll('.n-transfer-list-header__header')[1].text()).toBe(
      'test-target'
    )
  })
})
