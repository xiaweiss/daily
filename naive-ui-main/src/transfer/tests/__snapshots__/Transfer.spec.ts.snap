// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-transfer should work with \`size\` prop 1`] = `"--bezier: cubic-bezier(.4, 0, .2, 1); --bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --bezier-ease-out: cubic-bezier(0, 0, .2, 1); --border-color: rgb(224, 224, 230); --border-radius: 3px; --extra-font-size: 12px; --filter-divider-color: rgb(224, 224, 230); --font-size: 14px; --header-color: rgba(250, 250, 252, 1); --header-extra-text-color: rgb(51, 54, 57); --header-font-weight: 400; --header-text-color: rgb(31, 34, 37); --header-text-color-disabled: rgba(194, 194, 194, 1); --item-color-pending: rgb(243, 243, 245); --item-height: 28px; --item-text-color: rgb(51, 54, 57); --item-text-color-disabled: rgba(194, 194, 194, 1); --list-color: #fff; --width: 440px; --icon-color: rgba(194, 194, 194, 1); --icon-color-disabled: rgba(209, 209, 209, 1);"`;

exports[`n-transfer should work with \`size\` prop 2`] = `"--bezier: cubic-bezier(.4, 0, .2, 1); --bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --bezier-ease-out: cubic-bezier(0, 0, .2, 1); --border-color: rgb(224, 224, 230); --border-radius: 3px; --extra-font-size: 12px; --filter-divider-color: rgb(224, 224, 230); --font-size: 14px; --header-color: rgba(250, 250, 252, 1); --header-extra-text-color: rgb(51, 54, 57); --header-font-weight: 400; --header-text-color: rgb(31, 34, 37); --header-text-color-disabled: rgba(194, 194, 194, 1); --item-color-pending: rgb(243, 243, 245); --item-height: 34px; --item-text-color: rgb(51, 54, 57); --item-text-color-disabled: rgba(194, 194, 194, 1); --list-color: #fff; --width: 440px; --icon-color: rgba(194, 194, 194, 1); --icon-color-disabled: rgba(209, 209, 209, 1);"`;

exports[`n-transfer should work with \`size\` prop 3`] = `"--bezier: cubic-bezier(.4, 0, .2, 1); --bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --bezier-ease-out: cubic-bezier(0, 0, .2, 1); --border-color: rgb(224, 224, 230); --border-radius: 3px; --extra-font-size: 12px; --filter-divider-color: rgb(224, 224, 230); --font-size: 15px; --header-color: rgba(250, 250, 252, 1); --header-extra-text-color: rgb(51, 54, 57); --header-font-weight: 400; --header-text-color: rgb(31, 34, 37); --header-text-color-disabled: rgba(194, 194, 194, 1); --item-color-pending: rgb(243, 243, 245); --item-height: 40px; --item-text-color: rgb(51, 54, 57); --item-text-color-disabled: rgba(194, 194, 194, 1); --list-color: #fff; --width: 440px; --icon-color: rgba(194, 194, 194, 1); --icon-color-disabled: rgba(209, 209, 209, 1);"`;

